src/java.base/share/classes/java/lang/classfile/Attributes.java

```diff
@@ -207,836 +149,291 @@ public class Attributes {
-                }
-
-                @Override
-                protected void writeBody(BufWriter buf, SourceFileAttribute attr) {
-                    buf.writeIndex(attr.sourceFile());
-                }
-
-                @Override
-                public AttributeMapper.AttributeStability stability() {
-                    return AttributeStability.CP_REFS;
-                }
-            };
-
-    /** Attribute mapper for the {@code SourceID} attribute */
-    public static final AttributeMapper<SourceIDAttribute>
-            SOURCE_ID = new AbstractAttributeMapper<>(NAME_SOURCE_ID) {
-                @Override
-                public SourceIDAttribute readAttribute(AttributedElement e, ClassReader cf, int p) {
-                    return new BoundAttribute.BoundSourceIDAttribute(cf, this, p);
-                }
-
-                @Override
-                protected void writeBody(BufWriter buf, SourceIDAttribute attr) {
-                    buf.writeIndex(attr.sourceId());
-                }
-
-                @Override
-                public AttributeMapper.AttributeStability stability() {
-                    return AttributeStability.CP_REFS;
-                }
-            };
-
-    /** Attribute mapper for the {@code StackMapTable} attribute */
-    public static final AttributeMapper<StackMapTableAttribute>
-            STACK_MAP_TABLE = new AbstractAttributeMapper<>(NAME_STACK_MAP_TABLE) {
-                @Override
-                public StackMapTableAttribute readAttribute(AttributedElement e, ClassReader cf, int p) {
-                    return new BoundAttribute.BoundStackMapTableAttribute((CodeImpl)e, cf, this, p);
-                }
-
-                @Override
-                protected void writeBody(BufWriter b, StackMapTableAttribute attr) {
-                    StackMapDecoder.writeFrames(b, attr.entries());
-                }
-
-                @Override
-                public AttributeMapper.AttributeStability stability() {
-                    return AttributeStability.LABELS;
-                }
-            };
-
-
-    /** Attribute mapper for the {@code Synthetic} attribute */
-    public static final AttributeMapper<SyntheticAttribute>
-            SYNTHETIC = new AbstractAttributeMapper<>(NAME_SYNTHETIC, true) {
-                @Override
-                public SyntheticAttribute readAttribute(AttributedElement e, ClassReader cf, int p) {
-                    return new BoundAttribute.BoundSyntheticAttribute(cf, this, p);
-                }
-
-                @Override
-                protected void writeBody(BufWriter buf, SyntheticAttribute attr) {
-                    // empty
-                }
-
-                @Override
-                public AttributeMapper.AttributeStability stability() {
-                    return AttributeStability.STATELESS;
-                }
-            };
+    /**
+     * {@return Attribute mapper for the {@code AnnotationDefault} attribute}
```

Comment:
Just wondering, can we change `{@code AnnotationDefault}` to `{@value #NAME_ANNOTATION_DEFAULT}`, etc? This way, the names are still rendered as code in Javadoc HTML, but they are generated with links to the constants, and programmers will see these constants and prefer them over hardcoded values.

Reply:
On the other side it is questionable if the attribute names should be exposed in the API. We provide corresponding mappers and attribute models. I don't see a case where user would need to use the attribute names directly.

Reply:
Makes sense, we can always add these literals back if we do need them.